src/components/Insertion/index.tsx

```diff
@@ -0,0 +1,216 @@
+import { useState } from 'react';
+import { getDatabase, ref, set, push } from 'firebase/database';
+import app from '../../firebaseConfig';
+
+const Insertion: React.FC = () => {
+  const [insertionData, setInsertionData] = useState({
```

Comment:
direi che e' il momenti di introdurti alle librerie per i form

in questo caso sei riuscito a completare il form usando solo uno state, che ci sta, pero ti accorgerai subito delle limitazioni quando iniziano a servirti cose tipo validazione (es. nome non puo essere piu corto di 6 caratteri). 

la libreria piu comune di gran lunga adesso e' https://www.react-hook-form.com/

ti direi di installarla e migrare il form che hai fatto per utilizzare quella anziche lo state. solo a livello di logica, a livello di html sei messo bene non c'e bisogno che fai niente

Reply:
Perfetto, allora mi leggo la documentazione e cerco di modificare usando questa libreria